purview/api/v1.py

```diff
@@ -103,4 +122,110 @@ def get_resource_story(resource, uid):
+    """
+    item = get_neo4j_node(resource, uid)
+    if not item:
+        raise NotFound('This item does not exist')
+
+    uid_name = item.unique_id_property
+    forward_query = create_story_query(item, uid_name, uid)
+    backward_query = create_story_query(item, uid_name, uid, reverse=True)
+
+    def _get_partial_stories(query, resources_to_expand=None):
+        if resources_to_expand is None:
+            resources_to_expand = []
+
+        results_list = []
+        results, _ = db.cypher_query(query)
+
+        if not results:
+            return [results_list]
+
+        # Creating a list of lists where each list is a collection of node ids
+        # of the nodes present in that particular story path.
+        # Paths are re-sorted in the ascending order for simplifying the logic below
+        path_nodes_id = []
+        for path in reversed(results):
+            path_nodes_id.append([node.id for node in path[0].nodes])
+
+        unique_paths = []
+        for index, node_set in enumerate(path_nodes_id[:-1]):
+            unique = True
+            for alternate_set in path_nodes_id[index + 1:]:
+                # If the node_set is a subset of alternate_set,
+                # we know they are the same path except the alternate_set is longer.
+                # If alternate_set and node_set only have one node_id of difference,
+                # we know it's the same path but from the perspective of different siblings.
+                if set(node_set).issubset(set(alternate_set)) or len(
+                        set(alternate_set).difference(set(node_set))) == 1:
+                    unique = False
+                    break
+            if unique:
+                # Since results is from longest to shortest, we need to get
+                # the opposite index.
+                unique_paths.append(results[(len(path_nodes_id) - index) - 1][0])
+        # While traversing, the outer for loop only goes until the second to last element
+        # because the inner for loop always starts one element ahead of the outer for loop.
+        # Hence, all the subsets of the last element will not be added to the unique_paths
+        # list as the for loops will eliminate them. So we add the last element
+        # since we are sure it is unique.
+        unique_paths.append(results[0][0])
+        unique_paths_nodes = [path.nodes for path in unique_paths]
+
+        return PurviewStructuredNode.inflate_results(unique_paths_nodes, resources_to_expand)
+
+    if forward_query:
+        results_unordered_forward = _get_partial_stories(forward_query, [resource])
+    else:
+        results_unordered_forward = []
+
+    if backward_query:
+        results_unordered_backward = _get_partial_stories(backward_query, [resource])
+    else:
+        results_unordered_backward = []
+
+    all_results = []
+    if len(results_unordered_backward) == 0 or len(results_unordered_forward) == 0:
```

Comment:
Optional: You could just do:
`if not results_unordered_backward or not results_unordered_forward:`

Reply:
It works for me:
```python
>>> results_unordered_backward = []
>>> results_unordered_forward = [1, 2, 3]
>>> if not results_unordered_backward or not results_unordered_forward:
...     print('hello')
... 
hello
```

Reply:
oops! looks like I had a typo.